In [11]:
import pandas as pd

In [12]:
df_train = pd.read_csv('../data/train.csv')
df_train.head()

id  discourse_id  discourse_start  discourse_end  \
0  423A1CA112E2  1.622628e+12              8.0          229.0   
1  423A1CA112E2  1.622628e+12            230.0          312.0   
2  423A1CA112E2  1.622628e+12            313.0          401.0   
3  423A1CA112E2  1.622628e+12            402.0          758.0   
4  423A1CA112E2  1.622628e+12            759.0          886.0   

                                      discourse_text discourse_type  \
0  Modern humans today are always on their phone....           Lead   
1  They are some really bad consequences when stu...       Position   
2  Some certain areas in the United States ban ph...       Evidence   
3  When people have phones, they know about certa...       Evidence   
4  Driving is one of the way how to get around. P...          Claim   

  discourse_type_num                                   predictionstring  
0             Lead 1  1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 1...  
1         Position 1       45 46 47 48 49 50 51 52 53 54 55 56 57 58 59  
2         Evidence 1    60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75  
3         Evidence 2  76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 9...  
4            Claim 1  139 140 141 142 143 144 145 146 147 148 149 15...

In [13]:
df_train['id'].unique().shape

(15594,)

In [14]:
# measure number of words for discource text
df_train['discourse_text_len'] = df_train['discourse_text'].apply(lambda x: len(x.split()))
df_train['discourse_text_len'].describe()

count    144293.000000
mean         43.416569
std          43.163283
min           1.000000
25%          15.000000
50%          27.000000
75%          59.000000
max         836.000000
Name: discourse_text_len, dtype: float64

43 words on average, this is about very short texts. But note that this is about the discource text, not about the whole essay. How long are the essays?

In [15]:
import os
import numpy as np

directory = '../data/train/'

essays_number_of_words = []
for file in os.listdir(directory):
    # print(f"Reading file {directory} / {file}")
    with open(directory + file, 'r') as f:
        file_contents = f.read()
        num_words = len(file_contents.split())
        essays_number_of_words.append(num_words)
        

print(f"Average number of words : {np.mean(essays_number_of_words)}")

    

Average number of words : 421.2580479671669


And what part of these 421 essays is labelled as discourse element?

In [16]:
# should have done the above in pandas :)
# create a new column that contains the information about the number of words in the essay

essays_number_of_words = []
ctn =0
for file in os.listdir(directory):
    ctn += 1
    print(f"Reading file {ctn} / {len(os.listdir(directory))} : {directory} / {file}")
    with open(directory + file, 'r') as f:
        file_contents = f.read()
        num_words = len(file_contents.split())
        pandas_id = file.split('.')[0]
        # print(pandas_id)
    df_train.at [ df_train [ df_train['id']==pandas_id].index, 'essay_num_words'] = num_words

Reading file 1 / 15594 : ../data/train/ / 3321A3E87AD3.txt
Reading file 2 / 15594 : ../data/train/ / DFEAEC512BAB.txt
Reading file 3 / 15594 : ../data/train/ / 2E4AFCD3987F.txt
Reading file 4 / 15594 : ../data/train/ / EB6C2AF20BFE.txt
Reading file 5 / 15594 : ../data/train/ / A91A08E523D5.txt
Reading file 6 / 15594 : ../data/train/ / 616F8E0EFABF.txt
Reading file 7 / 15594 : ../data/train/ / 4E078101C9DD.txt
Reading file 8 / 15594 : ../data/train/ / B934F3106FCB.txt
Reading file 9 / 15594 : ../data/train/ / 05BE5364C50D.txt
Reading file 10 / 15594 : ../data/train/ / 96AC736B550F.txt
Reading file 11 / 15594 : ../data/train/ / F9368F6BAB62.txt
Reading file 12 / 15594 : ../data/train/ / B0CCFA21F82C.txt
Reading file 13 / 15594 : ../data/train/ / 617D56A15483.txt
Reading file 14 / 15594 : ../data/train/ / 8B5B15211390.txt
Reading file 15 / 15594 : ../data/train/ / F27691AEE065.txt
Reading file 16 / 15594 : ../data/train/ / BDF7794E56BF.txt
Reading file 17 / 15594 : ../data/train/ / F83702

In [20]:
df_train.to_csv("../data/train_with_num_words.csv", index=False)

ModuleNotFoundError: No module named 'pandas.io.formats.csvs'

In [18]:
df_train.to_csv("train_ext.csv")

ModuleNotFoundError: No module named 'pandas.io.formats.csvs'

In [17]:
df_train.groupby(['id'])["discourse_text_len"].sum()

id
0000D23A521A    251
00066EA9880D    622
000E6DE9E817    245
001552828BD0    512
0016926B079C    252
               ... 
FFF1442D6698    615
FFF1ED4F8544    412
FFF868E06176    419
FFFD0AF13501    204
FFFF80B8CC2F    168
Name: discourse_text_len, Length: 15594, dtype: int64

In [36]:
df = pd.DataFrame([ [1,2], [3,4], [5,6] , [7,8] ], columns=['max_speed', 'shield'])
df

max_speed  shield
0          1       2
1          3       4
2          5       6
3          7       8

In [39]:
mask = df[df['max_speed']==1]
# df["new_c"] = np.nan 
# df.loc[mask,"new_c"] = 123
mask.index

Int64Index([0], dtype='int64')

In [46]:
df.at[1,'new_c'] = 555
df[df['max_speed']==7].index

Int64Index([3], dtype='int64')

In [47]:
df.at[df[df['max_speed']==7].index, 'new_c'] = 777
df

max_speed  shield  new_c
0          1       2    NaN
1          3      55  555.0
2          5       6    NaN
3          7       8  777.0